R/pkg/inst/worker/daemon.R

```diff
@@ -47,8 +55,6 @@ while (TRUE) {
       close(inputCon)
       Sys.setenv(SPARKR_WORKER_PORT = port)
       try(source(script))
-      # Set SIGUSR1 so that child can exit
-      tools::pskill(Sys.getpid(), tools::SIGUSR1)
       parallel:::mcexit(0L)
```

Comment:
BTW, up to my knowledge, this should work alone, at least in C (< I didn't test). Even if it was a bug in R and is fixed in the future, I guess the current logics would still work with the current status.

```R
arbitrary <- 9999
tools::pskill(arbitrary, tools::SIGUSR1)
```

https://stat.ethz.ch/R-manual/R-devel/library/tools/html/pskill.html

> it silently ignores invalid values of its arguments, including zero or negative pids.